### PR TITLE
React/input currency pass event to callback

### PR DESCRIPTION
**Pull request by clairesunstudio**
This pr allow passing events to the onChange callout function, so that it can differentiate how the value is getting changed, get the type of events applied to InputCurrency.
It also fixes up/down button NaN value when default is null
screen shot 2019-03-04 at 4 03 50 pm

This pr is to fix the bug in PFML Benefits Calculator:
`npm link` to [calculator pr](https://github.com/massgov/calculators/pull/50)

### CODE REVIEW

**Comment by clairesunstudio** [comment]
This pr added parameter event type (and direction if any) to the InputCurrency onChange prop:
`onChange(newValue, props.id)` => `onChange(newValue, props.id, type, direction)`

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 1]
```diff
@@ -70,11 +71,12 @@ const Currency = (props) => (
             }
             context.updateState(update, () => {
               if (typeof props.onChange === 'function') {
-                props.onChange(numberValue, props.id);
+                props.onChange(e, numberValue, props.id);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                props.onChange(numberValue, props.id, e);
```

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 2]
```diff
@@ -94,7 +96,7 @@ const Currency = (props) => (
               const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
               context.updateState({ showError, errorMsg, value: toCurrency(newValue, 2) }, () => {
                 if (typeof props.onChange === 'function') {
-                  props.onChange(newValue, props.id);
+                  props.onChange(e, newValue, props.id);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                  props.onChange(newValue, props.id, e);
```

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 3]
```diff
@@ -144,29 +146,32 @@ const Currency = (props) => (
...
               if (!Number.isNaN(numberValue) && stringValue.length > 0) {
                 let newValue;
                 if (e.key === 'ArrowDown') {
-                  newValue = Number(Number.parseFloat(numberValue - props.step).toFixed(2));
+                  newValue = Number(Number.parseFloat(baseValue - props.step).toFixed(2));
                   const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
                   context.updateState({ showError, errorMsg, value: toCurrency(newValue, 2) }, () => {
                     if (typeof props.onChange === 'function') {
-                      props.onChange(newValue, props.id);
+                      props.onChange(e, newValue, props.id);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                      props.onChange(newValue, props.id, e);
```

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 4]
```diff
@@ -144,29 +146,32 @@ const Currency = (props) => (
...
                     }
                   });
                 } else if (e.key === 'ArrowUp') {
-                  newValue = Number(Number.parseFloat(numberValue + props.step).toFixed(2));
+                  newValue = Number(Number.parseFloat(baseValue + props.step).toFixed(2));
                   const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
                   context.updateState({ showError, errorMsg, value: toCurrency(newValue, 2) }, () => {
                     if (typeof props.onChange === 'function') {
-                      props.onChange(newValue, props.id);
+                      props.onChange(e, newValue, props.id);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                      props.onChange(newValue, props.id, e);
```

**Review by avrilpearl — APPROVED** [review]

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 5]
```diff
@@ -144,29 +146,32 @@ const Currency = (props) => (
...
                 let newValue;
-                if (e.key === 'ArrowDown') {
-                  newValue = Number(Number.parseFloat(numberValue - props.step).toFixed(2));
+                if (key === 'ArrowDown') {
+                  newValue = Number(Number.parseFloat(baseValue - props.step).toFixed(2));
                   const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
                   context.updateState({ showError, errorMsg, value: toCurrency(newValue, 2) }, () => {
                     if (typeof props.onChange === 'function') {
-                      props.onChange(newValue, props.id);
+                      props.onChange(newValue, props.id, type);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                      props.onChange(newValue, props.id, type, key);
```

Comment (clairesunstudio) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
                      props.onChange(newValue, props.id, type, key);
```

**Review by smurrayatwork — APPROVED** [review]

**Pull request merged**